src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1211,6 +1211,56 @@ class FunctionBinaryArithmetic : public IFunction
         return ColumnArray::create(res, typeid_cast<const ColumnArray *>(arguments[0].column.get())->getOffsetsPtr());
     }
 
+    ColumnPtr executeArrayWithNumericImpl(const ColumnsWithTypeAndName & args, const DataTypePtr & result_type, size_t input_rows_count) const
+    {
+        ColumnsWithTypeAndName arguments = args;
+        if (isNumber(args[0].type)) /// We should have scalar as a second argument
+            std::swap(arguments[0], arguments[1]);
+
+        const auto * return_type_array = checkAndGetDataType<DataTypeArray>(result_type.get());
+        if (!return_type_array)
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Return type for function {} must be array", getName());
+
+        auto num_args = arguments.size();
+        DataTypes data_types;
+
+        ColumnsWithTypeAndName new_arguments {num_args};
+        DataTypePtr result_array_type;
+
+        const auto * left_const = typeid_cast<const ColumnConst *>(arguments[0].column.get());
+        const auto * right_const = typeid_cast<const ColumnConst *>(arguments[1].column.get());
+
+        /// Unpacking arrays if both are constants.
+        if (left_const)
+        {
+            new_arguments[0] = {left_const->getDataColumnPtr(), arguments[0].type, arguments[0].name};
+            new_arguments[1] = {arguments[1].column.get()->getPtr(), arguments[1].type, arguments[1].name};
+            auto col = executeImpl(new_arguments, result_type, 1);
+            return ColumnConst::create(std::move(col), input_rows_count);
+        }
+
+        const auto * left_array_col = typeid_cast<const ColumnArray *>(arguments[0].column.get());
+        const auto & left_array_type = typeid_cast<const DataTypeArray *>(arguments[0].type.get())->getNestedType();
+        const auto & right_array_col = arguments[1].column.get()->cloneResized(left_array_col->size());
+
+        new_arguments[0] = {left_array_col->getDataPtr(), left_array_type, arguments[0].name};
+        if (right_const)
+            new_arguments[1] = {right_array_col->getPtr(), arguments[1].type.get()->getPtr(), arguments[1].name};
```

Comment:
Here `right_array_col->cloneResized(rows_count)` should be better than `right_array_col->getPtr()`. Because otherwise ColumnConst would have invalid size().

Also, `arguments[1].type.get()->getPtr()` -> `arguments[1].type`